system/jlib/jstats.cpp

```diff
@@ -1967,6 +2034,43 @@ class CStatisticCollection : public CInterfaceOf<IStatisticCollection>
             iter.query().serialize(out);
     }
 
+    virtual void deserialize(MemoryBuffer & in, unsigned version, int minDepth, int maxDepth) override
+    {
+        unsigned numStats;
+        in.read(numStats);
+        stats.ensureCapacity(numStats);
+        while (numStats-- > 0)
+        {
+            Statistic next(in, version);
+            if (minDepth <= 0)
+                stats.append(next);
+        }
+        unsigned numChildren;
+        in.read(numChildren);
+        children.ensure(numChildren);
+        while (numChildren-- > 0)
+        {
+            byte kind;
+            in.read(kind);
+            StatsScopeId childId;
+            childId.deserialize(in, version);
+            deserializeChild(childId, in, version, minDepth, maxDepth);
+        }
+    }
+
+    virtual void deserializeChild(const StatsScopeId & childId, MemoryBuffer & in, unsigned version, int minDepth, int maxDepth) override
+    {
+        if (maxDepth > 0)
+        {
+            IStatisticCollection * childCollection = ensureSubScope(childId, true);
+            childCollection->deserialize(in, version, (minDepth-1), (maxDepth-1));
+        }
+        else
+        {
+            deserializeNoStats(in, version);
```

Comment:
This seems a shame.  Can it be avoided for the common case i.e. where you are only reading the subgraph stats which are the root for hthor/thor graphs.